hack/generate/changelog/util/fragment_test.go

```diff
@@ -370,3 +374,46 @@ func TestGitPullRequestNumberGetter_parsePRNumber(t *testing.T) {
 		})
 	}
 }
+
+func TestGitPullRequestNumberGetter_getCommitMessage(t *testing.T) {
+	execCommand = fakeExecCommand
+	defer func() { execCommand = exec.Command }()
+
+	prg := &gitPullRequestNumberGetter{}
+	testCases := []struct {
+		file string
+		name string
+	}{
+		{
+			file: "testdata/9999-validfile.yaml",
+			name: "test getCommitMessage",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			commitStr, err := prg.getCommitMessage(tc.file)
+			assert.NoError(t, err)
+			assert.Contains(t, commitStr, mockedCommitStr)
+		})
+	}
+}
+
+func fakeExecCommand(command string, args ...string) *exec.Cmd {
+	mockedCommitStr = "sample PR (9999)"
+	cs := []string{"-test.run=TestExecCommandHelper", "--", command}
+	cs = append(cs, args...)
+	cmd := exec.Command(os.Args[0], cs...)
+	cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1",
+		"COMMIT_MSG=" + mockedCommitStr + "\n",
+	}
+	return cmd
+}
+
+func TestExecCommandHelper(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
+	}
+	//nolint:lll
+	fmt.Fprintln(os.Stdout, os.Getenv("COMMIT_MSG"))
+}
```

Comment:
as we check together. 

```
func fakeExecCommand(command string, args ...string) *exec.Cmd {
	cs := []string{"-test.run=TestExecCommandHelper", "--", command}
	cmd := exec.Command(os.Args[0], cs...)
	return cmd
}

func TestExecCommandHelper(t *testing.T) {
	fmt.Sprintln(mockedCommitStr)
}

```

Reply:
I'm having trouble following what this is doing. My initial thought would have been to see something like this:
```go
func fakeExecCommand(command string, _ ...string) *exec.Cmd {
	mockedCommitStr = "sample PR (9999)"
	cmd := exec.Command("echo", mockedCommitStr)
	return cmd
}
```

But it seems like I'm missing a key detail. Maybe add some comments to explain this?

Reply:
I agree with you Joe. The previous test is pretty tricky. I was trying to follow standard Golang's way of mocking exec.Command. What you suggested accomplishes the same without complicating it :).  I fixed it based on your suggestion.